tests/check_docs.mts

```diff
@@ -0,0 +1,163 @@
+const dirname = path.dirname(filename);
+
+interface NewerFile {
+  file: string;
+  commit: string;
+}
+
+type DocError =
+  | {
+      type: "out-of-date";
+      doc: string;
+      newerFiles: NewerFile[];
+    }
+  | {
+      type: "missing-front-matter";
+      doc: string;
+    }
+  | {
+      type: "missing-updated-field";
+      doc: string;
+    }
+  | {
+      type: "missing-files";
+      doc: string;
+    };
+
+const checkDoc = async (doc: string): Promise<DocError | undefined> => {
+  const docMatter = matter(await fs.readFile(doc));
+
+  if (!docMatter.data.updated) {
+    return {
+      type: "missing-updated-field",
+      doc,
+    };
+  }
+
+  // First: figure out how recent the documentation is by looking at the last
+  // commit to the file.
+  const { stdout: fileRevision } = await execFile("git", [
+    "rev-list",
+    "-1",
+    "HEAD",
+    "--",
+    doc,
+  ]);
+  const docRevision = fileRevision.trim();
+
+  // Next: figure out if any of the referenced files are more recent than our
+  // doc revision
+  const files: string[] = docMatter.data.files;
+  if (
+    !files ||
+    !Array.isArray(files) ||
+    !files.every((file) => typeof file === "string")
+  ) {
+    return {
+      type: "missing-files",
+      doc,
+    };
+  }
+
+  const fileChecks = await Promise.all(
+    files.map(async (file) => {
+      const { stdout: updated } = await execFile("git", [
+        "rev-list",
+        "-1",
+        "HEAD",
+        `^${docRevision}`,
+        "--",
+        file,
+      ]);
+      if (updated.trim().length > 0) {
+        return { file, commit: updated.trim() };
+      }
+
+      return undefined;
+    }),
+  );
+  const newerFiles = fileChecks.filter(
+    (fc): fc is NewerFile => fc !== undefined,
+  );
+  if (newerFiles.length > 0) {
+    return {
+      type: "out-of-date",
+      doc,
+      newerFiles,
+    };
+  }
+
+  return undefined;
+};
+
+const main = async () => {
+  const docsDir = path.join(dirname, "..", "docs");
+  const docs = await fs.readdir(docsDir);
+
+  const results = await Promise.all(
+    docs.map((doc) => checkDoc(path.join(docsDir, doc))),
+  );
+  const errors = results.filter(
+    (result): result is DocError => result !== undefined,
+  );
+
+  if (errors.length > 0) {
+    process.stderr.write("Documentation errors:\n\n");
+    for (const error of errors) {
+      switch (error.type) {
+        case "out-of-date":
+          process.stderr.write(
+            `  ${path.basename(error.doc)} is out of date\n`,
+          );
+          for (const newerFile of error.newerFiles) {
+            process.stderr.write(
+              `    ${newerFile.file} was updated in ${newerFile.commit}\n`,
+            );
+          }
+          process.stderr.write(
+            `  To fix this, either update ${path.basename(
+              error.doc,
+            )} or, if no changes are needed, change the updated field to match the most recent commit to those files.\n`,
```

Comment:
I presume we don't actually intend to list commit IDs in the updated field any more?  Maybe update the guidance given?